### ISSUE TITLE
parseable output format --json or so ?

### ISSUE THREAD

**Issue by grosser** [rikatz/kubepug#68]
we have lots of teams on our cluster, so having a way to get back "these 2 deployments are deprecated" in json would enable us to lookup up which team owns them and then send out notifications

**Comment by rikatz** [comment]
@grosser sorry, I'm a little bit slow this morning :)

Can you provide what output you expect?

Thank you!

**Comment by grosser** [comment]
anything that is not plain-text, for example:
```
{
 "deprecated": [
    {
      "kind":"Deployment",
      "api":"v1/beta1",
      "kubernetes_version":"1.16"
    }
  ]
}
```

**Comment by rikatz** [comment]
the latest version supports json output, as the following: kubepug --format json
```json
{"DeprecatedAPIs":[{"Description":"ClusterRole is a cluster level, logical grouping of PolicyRules that can be referenced as a unit by a RoleBinding or ClusterRoleBinding. Deprecated in v1.17 in favor of rbac.authorization.k8s.io/v1 ClusterRole, and will no longer be served in v1.22.","Group":"rbac.authorization.k8s.io","Kind":"ClusterRole","Version":"v1beta1","Name":"clusterroles","Deprecated":true,"Items":[{"Scope":"GLOBAL","ObjectName":"admin","Namespace":""},{"Scope":"GLOBAL","ObjectName":"cluster-admin","Namespace":""},{"Scope":"GLOBAL","ObjectName":"edit","Namespace":""},{"Scope":"GLOBAL","ObjectName":"kindnet","Namespace":""},{"Scope":"GLOBAL","ObjectName":"local-path-provisioner-role","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:aggregate-to-admin","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:aggregate-to-edit","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:aggregate-to-view","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:auth-delegator","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:basic-user","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:certificates.k8s.io:certificatesigningrequests:nodeclient","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:certificates.k8s.io:certificatesigningrequests:selfnodeclient","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:attachdetach-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:certificate-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:clusterrole-aggregation-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:cronjob-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:daemon-set-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:deployment-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:disruption-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:endpoint-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:expand-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:generic-garbage-collector","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:horizontal-pod-autoscaler","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:job-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:namespace-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:node-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:persistent-volume-binder","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:pod-garbage-collector","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:pv-protection-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:pvc-protection-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:replicaset-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:replication-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:resourcequota-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:route-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:service-account-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:service-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:statefulset-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:ttl-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:coredns","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:csi-external-attacher","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:csi-external-provisioner","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:discovery","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:heapster","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:kube-aggregator","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:kube-controller-manager","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:kube-dns","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:kube-scheduler","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:kubelet-api-admin","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:node","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:node-bootstrapper","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:node-problem-detector","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:node-proxier","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:persistent-volume-provisioner","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:public-info-viewer","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:volume-scheduler","Namespace":""},{"Scope":"GLOBAL","ObjectName":"view","Namespace":""}]},{"Description":"Role is a namespaced, logical grouping of PolicyRules that can be referenced as a unit by a RoleBinding. Deprecated in v1.17 in favor of rbac.authorization.k8s.io/v1 Role, and will no longer be served in v1.22.","Group":"rbac.authorization.k8s.io","Kind":"Role","Version":"v1beta1","Name":"roles","Deprecated":true,"Items":[{"Scope":"OBJECT","ObjectName":"kubeadm:bootstrap-signer-clusterinfo","Namespace":"kube-public"},{"Scope":"OBJECT","ObjectName":"system:controller:bootstrap-signer","Namespace":"kube-public"},{"Scope":"OBJECT","ObjectName":"extension-apiserver-authentication-reader","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"kube-proxy","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"kubeadm:kubelet-config-1.16","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"kubeadm:nodes-kubeadm-config","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"system::leader-locking-kube-controller-manager","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"system::leader-locking-kube-scheduler","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"system:controller:bootstrap-signer","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"system:controller:cloud-provider","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"system:controller:token-cleaner","Namespace":"kube-system"}]},{"Description":"RoleBinding references a role, but does not contain it.  It can reference a Role in the same namespace or a ClusterRole in the global namespace. It adds who information via Subjects and namespace information by which namespace it exists in.  RoleBindings in a given namespace only have effect in that namespace. Deprecated in v1.17 in favor of rbac.authorization.k8s.io/v1 RoleBinding, and will no longer be served in v1.22.","Group":"rbac.authorization.k8s.io","Kind":"RoleBinding","Version":"v1beta1","Name":"rolebindings","Deprecated":true,"Items":[{"Scope":"OBJECT","ObjectName":"kubeadm:bootstrap-signer-clusterinfo","Namespace":"kube-public"},{"Scope":"OBJECT","ObjectName":"system:controller:bootstrap-signer","Namespace":"kube-public"},{"Scope":"OBJECT","ObjectName":"kube-proxy","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"kubeadm:kubelet-config-1.16","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"kubeadm:nodes-kubeadm-config","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"system::extension-apiserver-authentication-reader","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"system::leader-locking-kube-controller-manager","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"system::leader-locking-kube-scheduler","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"system:controller:bootstrap-signer","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"system:controller:cloud-provider","Namespace":"kube-system"},{"Scope":"OBJECT","ObjectName":"system:controller:token-cleaner","Namespace":"kube-system"}]},{"Description":"ClusterRoleBinding references a ClusterRole, but not contain it.  It can reference a ClusterRole in the global namespace, and adds who information via Subject. Deprecated in v1.17 in favor of rbac.authorization.k8s.io/v1 ClusterRoleBinding, and will no longer be served in v1.22.","Group":"rbac.authorization.k8s.io","Kind":"ClusterRoleBinding","Version":"v1beta1","Name":"clusterrolebindings","Deprecated":true,"Items":[{"Scope":"GLOBAL","ObjectName":"cluster-admin","Namespace":""},{"Scope":"GLOBAL","ObjectName":"kindnet","Namespace":""},{"Scope":"GLOBAL","ObjectName":"kubeadm:kubelet-bootstrap","Namespace":""},{"Scope":"GLOBAL","ObjectName":"kubeadm:node-autoapprove-bootstrap","Namespace":""},{"Scope":"GLOBAL","ObjectName":"kubeadm:node-autoapprove-certificate-rotation","Namespace":""},{"Scope":"GLOBAL","ObjectName":"kubeadm:node-proxier","Namespace":""},{"Scope":"GLOBAL","ObjectName":"local-path-provisioner-bind","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:basic-user","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:attachdetach-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:certificate-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:clusterrole-aggregation-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:cronjob-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:daemon-set-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:deployment-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:disruption-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:endpoint-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:expand-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:generic-garbage-collector","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:horizontal-pod-autoscaler","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:job-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:namespace-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:node-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:persistent-volume-binder","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:pod-garbage-collector","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:pv-protection-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:pvc-protection-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:replicaset-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:replication-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:resourcequota-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:route-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:service-account-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:service-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:statefulset-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:controller:ttl-controller","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:coredns","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:discovery","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:kube-controller-manager","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:kube-dns","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:kube-scheduler","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:node","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:node-proxier","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:public-info-viewer","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system:volume-scheduler","Namespace":""}]},{"Description":"DEPRECATED - This group version of PriorityClass is deprecated by scheduling.k8s.io/v1/PriorityClass. PriorityClass defines mapping from a priority class name to the priority integer value. The value can be any valid integer.","Group":"scheduling.k8s.io","Kind":"PriorityClass","Version":"v1beta1","Name":"priorityclasses","Deprecated":true,"Items":[{"Scope":"GLOBAL","ObjectName":"system-cluster-critical","Namespace":""},{"Scope":"GLOBAL","ObjectName":"system-node-critical","Namespace":""}]}],"DeletedAPIs":null}
```

It's not prettified, but it's parseable

**Comment by grosser** [comment]
awesome, thx!

**Comment by rikatz** [comment]
I'll close then, but if this doesn't fit what you think is needed please feel free to reopen!

**Comment by grosser** [comment]
hmmm wanted to try this, but there is no new tag and I need a binary to download :)

**Comment by rikatz** [comment]
This is already at https://github.com/rikatz/kubepug/releases/tag/v1.0.0

The exit code flag will be released at v1.0.1 tomorrow :D

**Comment by grosser** [comment]
ahhh I though you just added it, all good then :D
